pkg/tsdb/influxdb/model_parser.go

```diff
@@ -40,11 +47,30 @@ func (qp *InfluxdbQueryParser) Parse(model *simplejson.Json, dsInfo *models.Data
 		return nil, err
 	}
 
-	queryInterval := model.Get("interval").MustString("")
-	intervalMS := model.Get("intervalMs").MustInt(0)
-	parsedInterval, err := tsdb.GetIntervalFrom(dsInfo.TimeInterval, queryInterval, int64(intervalMS), time.Millisecond*1)
-	if err != nil {
-		return nil, err
+	interval := query.Interval
+
+	// in old-alert use-case query.Interval is zero.
+	// for this case, to be backawrd-compatible,
+	// we calculate the final alert based on:
+	// - time-range
+	// - min-interval in json
+	// - min-interval from data-source-config
+	if interval == 0*time.Nanosecond {
```

Comment:
Can we update old alert to populate query.Interval properly and get rid of this extra code in "every" data source?

Reply:
you're right, i created a POC at https://github.com/grafana/grafana/pull/38434 , i can simplify this PR afterwards.

Reply:
that PR is merged now, i will update this PR.